kong/runloop/handler.lua

```diff
@@ -206,6 +215,13 @@ local function register_events()
       log(DEBUG, "[events] Service updated, invalidating router")
       cache:invalidate("router:version")
     end
+
+    if data.operation == "create" or
+       data.operation == "update" then
+      if utils.hostname_type(data.entity.host) == "name" then
+        timer_at(0, warmup_hostname_dns_timer, data.entity.host)
+      end
+    end
```

Comment:
It is worth noting that this event will only trigger for the node handling the Service creation/update. Those worker events are not propagated to other nodes in the cluster. The benefits of this early DNS query are very limited in production deployments imho, and is mostly optimizing a development/testing use-case.